### PR TITLE
Templates feature add editing

### PR DESCRIPTION
**Pull request by hannamocha**

### CODE REVIEW

File: calliope_app/template/views.py [hunk 1]
```diff
@@ -426,14 +413,16 @@ def get_or_create_template_technologies(template_type_techs, model, template_typ
...
+def get_or_create_template_carriers(template_type_carriers, model):
     new_carriers = {}
     model_carriers = Carrier.objects.filter(model_id=model.id).all()
 
     for carrier in template_type_carriers:
         existing_carrier = model_carriers.filter(name=carrier['name']).first()
         if existing_carrier: 
-            new_carriers[existing_carrier.name] = existing_carrier
+            if existing_carrier.rate_unit != carrier['rate_unit'] or existing_carrier.quantity_unit != carrier['quantity_unit']:
+                message = "Error: Carrier already exists with this name but the units are different. Please remove the existing carrier named '" + carrier['name'] + "' before attempting to add this Node Group to the model again."
```
Comment (hannamocha) [inline calliope_app/template/views.py:424]: I was hoping to pass this error message to the front end but couldn't figure out how to do it, does anyone have experience or an example of how to do this?

**Comment by jgu2** [comment]
@hannamocha I merged the feature of @mahithapillodi about scenario name renaming into `dev` branch, which has `.po` and `.mo` files update. I think you both have conflicts.

You'll need to pull the updates and re-make/compile the messages on both fr and es.

**Comment by hannamocha** [comment]
@jgu2 Yes it's ready for review now, the function that calls that function catches that error and returns an http response now.

File: calliope_app/template/views.py [hunk 2]
```diff
@@ -125,77 +125,56 @@ def add_template(request):
...
-            Template.objects.filter(id=template_id).update(
+    try: 
+        comment = ""
+        template = {}
+        model_uuid = request.POST.get("model_uuid", False)
+        template_id = request.POST.get("template_id", False)
+        name = request.POST["name"]
+        template_type_id = request.POST["template_type"]
+        location_id = request.POST["location"]
+        varData = json.loads(request.POST["form_data"])
```
[the remark this answers is not in the document]
Comment (hannamocha) [reply calliope_app/template/views.py]: I ended up escaping the list items of dict since that recursive_escpae function wasn't set up to handle lists.

**Review by jgu2 — APPROVED** [review]
Looks good now, thanks for the update.

**Pull request merged**